src/streamlink/plugins/mbcmini.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+import logging
+import re
+import json
+
+from streamlink.plugin import Plugin
+from streamlink.plugin.api import useragents
+
+log = logging.getLogger(__name__)
+
+
+class Mbcmini(Plugin):
+
+    _url_re = re.compile(r'https?://miniplay\.imbc\.com/WebLiveURL\.ashx\?channel=(?P<channel_id>[a-z]+)')
+
+    _api_url = '''http://miniplay.imbc.com/WebLiveURL.ashx?channel={0}&protocol=M3U8\
+&agent=android&androidversion=18&callback=jarvis.miniInfo.loadOnAirComplete'''
+
+    @classmethod
+    def can_handle_url(cls, url):
+        return cls._url_re.match(url) is not None
+
+    def _get_streams(self):
+        match = self._url_re.match(self.url)
+        channel_id = match.group("channel_id")
+        json_url_key = "AACLiveURL"
+
+        headers = {
+            'User-Agent': useragents.CHROME,
+            'Referer': '''http://mini.imbc.com/webapp_v3/minifloat.html?\
+src=http://www.imbc.com/broad/radio/minimbc/index.html&ref='''
+        }
+        url = self._api_url.format(channel_id)
+
+        res = self.session.http.get(url, headers=headers)
+
+        json_text = '{' + ''.join(res.text.split()[1:-1])
+        json_data = json.loads(json_text)
```

Comment:
Plugins should use Streamlink's validator API to return results from network requests, if possible. The API response appears to be a JSONP call, which means you should extract the JSON payload via a regex instead of splitting the response text by newline characters.

Reply:
Modified to use Streamlink's validation API.